proposals/handler.go

```diff
@@ -481,11 +482,16 @@ func (h *Handler) checkBallotSyntacticValidity(
 	ballotDuration.WithLabelValues(fetchRef).Observe(float64(time.Since(t1)))
 
 	t2 := time.Now()
-	// ballot can be decoded only if all dependencies (blocks, ballots, atxs) were downloaded
+	// ballot can be decoded only if all dependencies (ballots, atxs) were downloaded
 	// and added to the tortoise.
 	decoded, err := h.tortoise.DecodeBallot(b.ToTortoiseData())
 	if err != nil {
-		return nil, fmt.Errorf("failed to decode ballot id %s err %w", b.ID().AsHash32().ShortString(), err)
+		return nil, fmt.Errorf(
+			"%w: failed to decode ballot id %s. %v",
```

Comment:
Since Go 1.20 it is possible to wrap multiple errors into one:

```suggestion
			"%w: failed to decode ballot id %s. %w",
```

Reply:
i remember but i don't find it useful, there are errors that simply add more context and errors that are a part of the api